streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -1694,6 +1694,28 @@ public void shouldMergeMultipleStreams() {
             processorSupplier.theCapturedProcessor().processed());
     }
 
+    @Test
+    public void shouldMergeSameStreams() {
+        final String topic1 = "topic-1";
+
+        final KStream<String, String> source = builder.stream(topic1);
```

Comment:
Instead of creating a new `KStream` we could just re-use the prepared `KStream testStream`
```
final KStream<String, String> merged = testStream.merge(testStream);
```